specification/cosmos-db/data-plane/Microsoft.TablesStorage/preview/2018-10-10/table.json

```diff
@@ -0,0 +1,1585 @@
+          },
+          {
+            "$ref": "#/parameters/selectParameter"
+          },
+          {
+            "$ref": "#/parameters/filterParameter"
+          },
+          {
+            "$ref": "#/parameters/tableNameParameter"
+          },
+          {
+            "$ref": "#/parameters/partitionKeyParameter"
+          },
+          {
+            "$ref": "#/parameters/rowKeyParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "headers": {
+              "x-ms-client-request-id": {
+                "x-ms-client-name": "ClientRequestId",
+                "type": "string",
+                "description": "If a client request id header is sent in the request, this header will be present in the response with the same value."
+              },
+              "x-ms-request-id": {
+                "x-ms-client-name": "RequestId",
+                "type": "string",
+                "description": "This header uniquely identifies the request that was made and can be used for troubleshooting the request."
+              },
+              "x-ms-version": {
+                "x-ms-client-name": "Version",
+                "type": "string",
+                "description": "Indicates the version of the Queue service used to execute the request. This header is returned for requests made against version 2009-09-19 and above."
+              },
+              "Date": {
+                "type": "string",
+                "format": "date-time-rfc1123",
+                "description": "UTC date/time value generated by the service that indicates the time at which the response was initiated"
+              },
+              "x-ms-continuation-NextPartitionKey": {
+                "type": "string",
+                "description": "This header contains the continuation token value for partition key."
+              },
+              "x-ms-continuation-NextRowKey": {
+                "type": "string",
+                "description": "This header contains the continuation token value for row key."
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/TableEntityQueryResponse"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "headers": {
+              "x-ms-error-code": {
+                "x-ms-client-name": "ErrorCode",
+                "type": "string"
+              }
+            },
+            "schema": {
+              "$ref": "#/definitions/StorageError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "table"
+        ],
+        "operationId": "Table_UpdateEntity",
+        "x-ms-examples": {
+          "TableUpdateEntity": {
+            "$ref": "./examples/TableUpdateEntity.json"
+          }
+        },
+        "description": "Update entity in a table.",
+        "parameters": [
```

Comment:
I believe we are missing a required `If-Match` header parameter here. In its current form this method implements 'InsertOrReplace'

We could either rename this one to that and implement Update with the `If-Match` header or vice versa.

Another option could be to make `If-Match` optional, comment appropriately in the description, and allow the implementation detail of implementing `Update` or 'InsertOrReplace' to the client SDK.